### DISCUSSION TITLE
How to approach existing libraries

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/membership#52] [General]
I'm curious if folks have any ideas on how we could or should approach existing libraries.

I view our value proposition as the last repo transfer you'll have to do because you'll know the package is permanently in a place that will have maintainers.

Now, that's not to say every package will be maintained, we need people to do that. However this org is focused on sustainability and reliability through redundancy from the top down. I think that's worth something and package maintainers will value that enough to spend the couple hours moving it over.

I guess I have two questions:
1. Does the above match your thoughts? 
2. Do you have ideas on how to phrase this to maintainers?

cc: @django-commons/admins

**Comment by ryancheley** [comment]
Your thoughts broadly match mine 👍 

I think that there are couple of ways that we can outreach to maintainers (which is how I’m interpreting your second question):

1. Broad Marketing
2. Direct outreach

## Broad Marketing with a Call to Action to reach out to us

- Appearing on podcasts like DjangoChat (which @tim-schilling will be doing on Oct 1, 2024)
- Blogging about Django Commons
- Conference Talks (either full talks, or lightening talks like @williln did at Django Con US 2024)
- work to get a section in the Django News Newsletter for projects that need contributors and have an end note of some kind mentioning Django Commons (I spoke with Will Vincent about the contributors needed section at DCUS 2024, but not the end note part)
- Sharing the videos and blog posts from above on social media through our personal accounts and an official Django-commons account when we have one

## Direct Outreach

- identify projects that seem to need help though I’m not sure how we do this necessarily or what ‘help’ necessary means in this context

These are just the first few things that come to mind. Given that it might be easier (at least initially) to do broad marketing to get the word out about what we’re doing that might be the best place to start and then we can revisit the direct outreach after we have maybe 10 or 15 ‘success’ stories

**Comment by williln** [comment]
I'm going to speak to the question about phrasing the value proposition for maintainers: 

- Focus on "extra people to have your back when life gets in the way of your project maintenance," "easing the burden of maintenance," resting assured that your package will be able to respond to critical bugs even if you need to step away for a life event, etc. 
- Additional focus on something like "potential maintainer matchmaking" -- we have and are always recruiting a team of potential contributors and maintainers. You let us know that you want some help on your package, and we can assist in finding someone to get you that help. 

The value proposition for users (and thus for maintainers who want increased adoption of their project) includes things like: 

- Projects in Django Commons meet some set criteria. They must have some tests, they must have a Code of Conduct, etc. 
- We don't want to over-promise. We aren't fully vetting every project and we don't pretend like we are. But there is _some_ vetting, and also a community effort to keep packages that are part of Django Commons up to date and free of major bugs. 

### Marketing 

- We talked about potentially creating a channel in the Django Discord, or using the Discussions feature here on GitHub, as a way to reach potential maintainers who might want to transfer projects in, and a way to reach potential contributors who need a project to work on. 
- When the DCUS lightning talks are released, we can extract my lightning talk and post it somewhere

**Reply by tim-schilling** [reply]
> We talked about potentially creating a channel in the Django Discord

There is the packages channel which is probably as good as we'll get. There's understandable hesitancy to add channels for all projects/initiatives

**Reply by ryancheley** [reply]
> There is the packages channel which is probably as good as we'll get. There's understandable hesitancy to add channels for all projects/initiatives

In that case I won’t reach out to any of the Discord admins to ask about it.

**Reply by nanorepublica** [reply]
You could reach out to the admins for an announcement on the discord server. This would ping those interested in hearing community announcements which this qualifies in my mind.

What might be worth asking about is a 'django-commons' role? This could be a pingable role which allows those interested in the server to make themselves known? This only works if though if there are those willing to get pinged for a conversation!

**Reply by ryancheley** [reply]
> You could reach out to the admins for an announcement on the discord server. This would ping those interested in hearing community announcements which this qualifies in my mind.
> 
> What might be worth asking about is a 'django-commons' role? This could be a pingable role which allows those interested in the server to make themselves known? This only works if though if there are those willing to get pinged for a conversation!

I’ve added this as a discussion point for our next admin meeting on Oct 18

**Reply by ryancheley** [reply]
The announcement has been made on the Django Discord Announcement channel 🎉

https://discord.com/channels/856567261900832808/857649871719759903/1300991918574141560

**Reply by tim-schilling** [reply]
Thanks Ryan!

**Comment by acdha** [comment]
If you wanted to trial the process, I'd like to transfer the projects under https://github.com/django-haystack and would be happy to run through the process as a pilot.

**Reply by ryancheley** [reply]
> If you wanted to trial the process, I'd like to transfer the projects under https://github.com/django-haystack and would be happy to run through the process as a pilot.

@acdha that’s great to hear! There is a page [here](https://github.com/django-commons/membership?tab=readme-ov-file#how-to-transfer-a-project-in) that talks about how to transfer a project in. So far we’ve had a couple of successful transfers

- [django-tasks-scheduler](https://github.com/django-commons/django-tasks-scheduler)
- [django-typer](https://github.com/django-commons/django-typer)
- [django-fam-2](https://github.com/django-commons/django-fsm-2)

If you’re interested in getting started we’d love to have you join!

**Reply by tim-schilling** [reply]
Hi @acdha! We've moved a couple over and happy to bring others into the fold as people want to transition them. Please don't do it solely for the purpose of being a pilot test. While we're still definitely learning, the process isn't entirely new anymore.

**Reply by acdha** [reply]
I'll look into that – I don't use those projects any more since I've shifted jobs and I'm really looking for a way to hand them over to someone I can trust not to be the next Jia Tan.

**Reply by tim-schilling** [reply]
Sounds good! As of right now, we require someone to signed on to actively maintain the library. We're still ramping up in the "find qualified new maintainers for projects" department. That is one of our goals though.
